client/context/helpers.go

```diff
@@ -185,6 +201,32 @@ func (ctx CoreContext) EnsureSignBuildBroadcast(name string, msgs []sdk.Msg, cdc
 	return ctx.BroadcastTx(txBytes)
 }
 
+// sign and build the transaction from the msg
+func (ctx CoreContext) EnsureSignBuildBroadcastAsync(name string, msgs []sdk.Msg, cdc *wire.Codec) (res *ctypes.ResultBroadcastTx, err error) {
+
```

Comment:
It looks like this is the same as EnsureSignBuildBroadcast, could we abstract the functionality of these two into another method, perhaps `ensureSignBuild` which takes the same parameters as these two, but returns `txBytes, err`, and if that doesn't error, then run the respective broadcast method on each? (Just to reduce code duplication. I know this will be changed whenever we change how we're getting sequence numbers for async tx's, but since async is primarily intended for testing, I think that will be awhile)